src/ajax.js

```diff
@@ -861,4 +861,14 @@ jQuery.each( [ "get", "post" ], function( _i, method ) {
 	};
 } );
 
+jQuery.ajaxPrefilter( function( s ) {
+	if ( s.data && s.processData ) {
```

Comment:
I don't think this check is needed. The value of `contentType` shouldn't depend on whether data is passed & being processed.

Reply:
You are right - done! (Also saves a few bytes.)